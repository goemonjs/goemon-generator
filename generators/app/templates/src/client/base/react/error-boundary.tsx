import React from 'react';
import { getLogger } from '../utilities/logger';

interface IState {
  hasError: boolean;
  errorMessage: string;
  errorInfo: string;
}

export class ErrorBoundary extends React.Component<{}, IState>  {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
      errorMessage: '',
      errorInfo: ''
    };
  }

  componentDidCatch(error, info) {
    this.setState({
      hasError: true,
      errorMessage: error.message,
      errorInfo: info.componentStack
    });
    if (typeof window !== 'undefined') {  // Check wheter this is client side
      let logger = getLogger();
      logger.error(error.message, info.componentStack);
    }
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          Application Error
        </div>
      );
    }
    return this.props.children;
  }
}
