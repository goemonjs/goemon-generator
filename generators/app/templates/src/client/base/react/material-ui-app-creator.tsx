// /*---------------------------------------------------------------------------------------------
//  *  Copyright (c) Lunascape Corporation. All rights reserved.
//  *--------------------------------------------------------------------------------------------*/
// import * as React from 'react';
// import { Route, Switch } from 'react-router';
// import { BrowserRouter, StaticRouter } from 'react-router-dom';
// import { renderRoutes } from 'react-router-config';
// import { Provider } from 'react-redux';
// import { renderToString } from 'react-dom/server';
// import { SheetsRegistry } from 'react-jss/lib/jss';
// import JssProvider from 'react-jss/lib/JssProvider';
// import { MuiThemeProvider, createMuiTheme, createGenerateClassName } from '@material-ui/core/styles';
// import { I18nextProvider } from 'react-i18next';
// import * as moment from 'moment';
// import MuiPickersUtilsProvider from 'material-ui-pickers/utils/MuiPickersUtilsProvider';
// import MomentUtils from 'material-ui-pickers/utils/moment-utils';

// import i18nClient from '../../../localization/i18n-client';

// class JSTUtils extends MomentUtils {

//   getHours(date) {
//     return date.utcOffset(9).get('hours');
//   }

//   getDatePickerHeaderText(date) {
//     return date.utcOffset(9).format('ddd, MMM D');
//   }

//   getDayText(date) {
//     return date.utcOffset(9).format('D');
//   }

//   getYearText(date) {
//     return date.utcOffset(9).format('YYYY');
//   }
// }

// export function createClientApp(componant, theme, store, basename, preloadedLanguage = 'en') {
//   // Create a sheetsRegistry instance.
//   const sheetsRegistry = new SheetsRegistry();

//   const generateClassName = createGenerateClassName();

//   i18nClient.changeLanguage(preloadedLanguage);

//   // Localization moment
//   moment.locale(preloadedLanguage);

//   return (
//     <JssProvider registry={sheetsRegistry} generateClassName={generateClassName}>
//       <I18nextProvider i18n={i18nClient}>
//         <MuiThemeProvider theme={theme}>
//           <Provider store={store}>
//             <BrowserRouter basename={basename}>
//               <MuiPickersUtilsProvider utils={JSTUtils}>
//                 {componant}
//               </MuiPickersUtilsProvider>
//             </BrowserRouter>
//           </Provider>
//         </MuiThemeProvider>
//       </I18nextProvider>
//     </JssProvider>
//   );
// }

// // Render the component to a string.
// export function renderOnServer(componant, theme, req, context, store, i18nServer) {

//   // Create a sheetsRegistry instance.
//   const sheetsRegistry = new SheetsRegistry();

//   const generateClassName = createGenerateClassName();

//   const html = renderToString(
//     <JssProvider registry={sheetsRegistry} generateClassName={generateClassName}>
//       <I18nextProvider i18n={i18nServer}>
//         <MuiThemeProvider theme={theme} sheetsManager={new Map()}>
//           <Provider store={store}>
//             <StaticRouter location={req.baseUrl + req.url} context={context}>
//               {componant}
//             </StaticRouter>
//           </Provider>
//         </MuiThemeProvider>
//       </I18nextProvider>
//     </JssProvider>
//   );

//   const css = sheetsRegistry.toString();

//   return {
//     html: html,
//     css: css
//   };
// }

//  // Grab the CSS from our sheetsRegistry.
