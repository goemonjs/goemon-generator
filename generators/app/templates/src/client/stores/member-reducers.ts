import { createTypeReducer } from 'type-redux';
import * as Actions from '../actions/profile-actions';

// Redux Action Reducer Samples
export type IState = {
  message: string;
  snackBarState: {
    type: 'success' | 'warning' | 'error' | 'info';
    open: boolean;
  };
  profile: {
    userid: string,
    username: string
  }
};

export const initialState: IState = {
  message: '',
  snackBarState: {
    type: 'info',
    open: false
  },
  profile: {
    userid : '',
    username : ''
  }
};

export const getProfileReducer = Actions.getProfile.reducer<IState>((state, action) => {
  if ( action.error ) {
    return {
      userid: '',
      username: '',
      message: action.payload!.message && <string>action.meta,
      snackBarState: {
        type: 'error',
        open: true
      }
    };
  }
  return {
    profile: {
      userid: action.payload.id,
      username: action.payload.email
    },
    message: ''
  };
});

export const reducer = createTypeReducer(
  initialState,
  getProfileReducer
);
