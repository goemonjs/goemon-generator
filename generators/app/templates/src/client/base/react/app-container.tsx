import * as React from 'react';
import { Provider } from 'react-redux';
import { UserContext, IContextProps } from '../../context/user-context';
import { BrowserRouter, StaticRouter } from 'react-router-dom';

interface IProps  {
  store: any;
  location?: string;
  context?: any;
  basename?: string;
}
export class AppContainer extends React.Component<IProps, {}> {
  render () {
    const { store, context, location, basename } = this.props;
    if ( typeof window !== 'undefined' ) { // Check whether this method is called on client or server
      return (
        <Provider store={store}>
          <BrowserRouter basename={basename}>
            <UserContext.Provider value={context}>
              {this.props.children}
            </UserContext.Provider>
          </BrowserRouter>
        </Provider>
      );
    } else {
      return (
        <Provider store={store}>
          <StaticRouter location={location} context={context}>
            {this.props.children}
          </StaticRouter>
        </Provider>
      );
    }
  }
}
