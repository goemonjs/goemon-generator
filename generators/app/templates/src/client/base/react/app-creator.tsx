/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import * as React from 'react';
import { BrowserRouter, StaticRouter, Route, Switch } from 'react-router-dom';
import { renderRoutes } from 'react-router-config';
import { Provider } from 'react-redux';
import { renderToString } from 'react-dom/server';

export function createClientApp(componant, store, basename) {
  return (
    <Provider store={store}>
      <BrowserRouter basename={basename}>
          {componant}
      </BrowserRouter>
    </Provider>
  );
}

// Render the component to a string.
export function renderOnServer(componant, req, context, store) {

  const html = renderToString(
    <Provider store={store}>
      <StaticRouter location={req.baseUrl + req.url} context={context}>
        {componant}
      </StaticRouter>
    </Provider>
  );

  return {
    html: html
  };
}

 // Grab the CSS from our sheetsRegistry.
