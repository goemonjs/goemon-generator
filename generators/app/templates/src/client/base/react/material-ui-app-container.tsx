import * as React from 'react';
import { SheetsRegistry } from 'react-jss/lib/jss';
import JssProvider from 'react-jss/lib/JssProvider';
import { MuiThemeProvider, createGenerateClassName } from '@material-ui/core/styles';
import { AppContainer } from './app-container';

interface IProps  {
  store: any;
  location?: string;
  context?: any;
  basename?: string;
  theme?: any;
  sheetsRegistry?: SheetsRegistry;
}
export class MaterialUiAppContainer extends React.Component<IProps, {}> {
  // Remove the server-side injected CSS.
  componentDidMount() {
    const jssStyles = document.getElementById('jss-server-side');
    if (jssStyles && jssStyles.parentNode) {
      jssStyles.parentNode.removeChild(jssStyles);
    }
  }

  render () {
    const { theme, sheetsRegistry } = this.props;

    const generateClassName = createGenerateClassName();

    if ( typeof window !== 'undefined' ) { // Check whether this method is called on client or server
      return (
        <JssProvider registry={sheetsRegistry} generateClassName={generateClassName}>
          <MuiThemeProvider theme={theme}>
            <AppContainer {...this.props} />
        </MuiThemeProvider>
      </JssProvider>
      );
    } else {
      return (
        <JssProvider registry={sheetsRegistry} generateClassName={generateClassName}>
          <MuiThemeProvider theme={theme} sheetsManager={new Map()}>
            <AppContainer {...this.props} />
          </MuiThemeProvider>
        </JssProvider>
      );
    }
  }
}
