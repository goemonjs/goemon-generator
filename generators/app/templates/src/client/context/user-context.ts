/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Lunascape Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/
import * as React from 'react';

export type IContextProps = {
  userType: 'guest' | 'member' | 'admin';
};

export const UserContext = React.createContext<IContextProps>({
  userType: 'guest'
});
