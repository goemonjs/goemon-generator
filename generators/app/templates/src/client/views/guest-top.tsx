import * as React from 'react';
import { SimpleList } from './components/simple-list';

interface IProps extends React.Props<{}> {
  message: string;
  style: any;
}

export class GuestTop extends React.Component<IProps, any> {
  render() {
    let { message, style } = this.props;
    return (
      <>
        <div className="jumbotron">
          <h1>Express-React-Redux-Typescript Sample</h1>
          <p className="lead">Hello world!</p>
          <a className="btn btn-large btn-success" href="/simple">React Sample</a>
          <a className="btn btn-large btn-success" href="/redux">Redux Sample</a>
          <a className="btn btn-large btn-success" href="/auth">Auth Sample</a>
        </div>
      </>
    );
  }
}
