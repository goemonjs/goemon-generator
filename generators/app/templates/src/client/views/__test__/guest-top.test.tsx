import * as React from 'react';
import renderer from 'react-test-renderer';

import { UserContext, IContextProps } from '../../context/user-context';
import { AppContainer } from '../../base/react/app-container';
import { ErrorBoundary } from '../../base/react/error-boundary';
import i18n from '../../localization/i18n';

import { configureStore } from '../../stores/guest-store';
import { GuestTop } from '../guest-top';

describe('Guest Top Component Test', () => {
  beforeAll(() => {
    let mockWindow: any = window;
    mockWindow.scrollTo = (x, y) => {
      console.log(`Scroll to (${x}, ${y})`);
    };
  });

  const store = configureStore();
  const userContext: IContextProps = {
    userType: 'guest',
  };

  test('renders correctly in English', () => {
    const i18nInstance = i18n.cloneInstance();
    i18nInstance.changeLanguage('en');
    const tree = renderer.create(
      <ErrorBoundary>
        <AppContainer i18n={i18nInstance} store={store} context={userContext}>
          <GuestTop />
        </AppContainer>
      </ErrorBoundary>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly in Japanese', () => {
    const i18nInstance = i18n.cloneInstance();
    i18nInstance.changeLanguage('ja');
    const tree = renderer.create(
      <ErrorBoundary>
        <AppContainer i18n={i18nInstance} store={store} context={userContext}>
          <GuestTop />
        </AppContainer>
      </ErrorBoundary>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly in not supported language', () => {
    const i18nInstance = i18n.cloneInstance();
    i18nInstance.changeLanguage('fr');
    const tree = renderer.create(
      <ErrorBoundary>
        <AppContainer i18n={i18nInstance} store={store} context={userContext}>
          <GuestTop />
        </AppContainer>
      </ErrorBoundary>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });


});
