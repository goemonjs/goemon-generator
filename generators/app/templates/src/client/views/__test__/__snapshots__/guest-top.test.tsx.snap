// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guest Top Component Test renders correctly in English 1`] = `
<div
  className="jumbotron"
>
  <h1>
    Goemon
  </h1>
  <p
    className="lead"
  >
    The boilerplate using React16, Redux, React-Router v4 + SSR(Server Side Rendering), Node, Express, and MongoDB in TypeScript.
  </p>
  <a
    className="btn btn-large btn-success"
    href="/react"
  >
    React Sample
  </a>
  <a
    className="btn btn-large btn-success"
    href="/redux"
  >
    Redux Sample
  </a>
  <div>
    <a
      href="/?lng=en"
    >
      English
    </a>
  </div>
  <div>
    <a
      href="/?lng=ja"
    >
      日本語
    </a>
  </div>
</div>
`;

exports[`Guest Top Component Test renders correctly in Japanese 1`] = `
<div
  className="jumbotron"
>
  <h1>
    五右衛門
  </h1>
  <p
    className="lead"
  >
    The boilerplate using React16, Redux, React-Router v4 + SSR(Server Side Rendering), Node, Express, and MongoDB in TypeScript.
  </p>
  <a
    className="btn btn-large btn-success"
    href="/react"
  >
    React Sample
  </a>
  <a
    className="btn btn-large btn-success"
    href="/redux"
  >
    Redux Sample
  </a>
  <div>
    <a
      href="/?lng=en"
    >
      English
    </a>
  </div>
  <div>
    <a
      href="/?lng=ja"
    >
      日本語
    </a>
  </div>
</div>
`;

exports[`Guest Top Component Test renders correctly in not supported language 1`] = `
<div
  className="jumbotron"
>
  <h1>
    Goemon-
  </h1>
  <p
    className="lead"
  >
    The boilerplate using React16, Redux, React-Router v4 + SSR(Server Side Rendering), Node, Express, and MongoDB in TypeScript.
  </p>
  <a
    className="btn btn-large btn-success"
    href="/react"
  >
    React Sample
  </a>
  <a
    className="btn btn-large btn-success"
    href="/redux"
  >
    Redux Sample
  </a>
  <div>
    <a
      href="/?lng=en"
    >
      English
    </a>
  </div>
  <div>
    <a
      href="/?lng=ja"
    >
      日本語
    </a>
  </div>
</div>
`;
